### PR TITLE
Issue 9818 caffine cache impl

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

**Comment by wezell** [comment]
Thanks for the heads up @ben-manes !  We have customers who don't ever size their cache and hit oom exceptions, which is painful for our support dept.  This is good information and will make the change.

File: src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        return "Caffine Memory Cache";
+    }
+
+    @Override
+    public String getKey() {
+        return "LocalCaffineMem";
+    }
+
+    @Override
+    public void init() {
```
Comment (dsilvam) [inline src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java]: There is at least two more CacheProviders (TimeCachedProvider, GuavaCache) with a very similar code, if not the same, for the init. Thoughts @freddyucv , @jdotcms, @agomez-dotcms, @jgambarios

File: src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        return list;
+    }
+
+    @Override
+    public void shutdown() {
+        Logger.info(this.getClass(), "===== Calling shutdown [" + getName() + "].");
+        isInitialized = false;
+    }
+
+    private Cache<String, Object> getCache(String cacheName) {
```
Comment (dsilvam) [inline src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java]: This method also shares a lot of code with other Providers (GuavaCache, TimedCacheProvider). Thoughts @freddyucv , @jdotcms, @agomez-dotcms, @jgambarios

File: src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java (hunk 1 above)
Comment (dsilvam) [inline src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java]: There is at least two more CacheProviders (TimeCachedProvider, GuavaCache) with a very similar code, if not the same, for the init. Thoughts @freddyucv , @jdotcms, @agomez-dotcms, @jgambarios

File: src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java (hunk 2 above)
Comment (dsilvam) [inline src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java]: This method also shares a lot of code with other Providers (GuavaCache, TimedCacheProvider). Thoughts @freddyucv , @jdotcms, @agomez-dotcms, @jgambarios

Comment (dsilvam) [inline src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java]: We got together and determined that refactoring to abstract the repeated code does not bring benefit. Leave as it is.

File: src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java (hunk 1 above)
Comment (dsilvam) [inline src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java]: We got together and determined that refactoring to abstract the repeated code does not bring benefit. Leave as it is.

**Review by jgambarios — APPROVED** [review]

**Pull request merged**